android/src/main/java/com/fidelreactlibrary/events/BridgeLibraryEventEmitter.java

```diff
@@ -1,22 +1,24 @@
 package com.fidelreactlibrary.events;
 
 import com.facebook.react.bridge.ReactApplicationContext;
-import com.facebook.react.bridge.WritableMap;
+import com.facebook.react.bridge.ReadableMap;
 import com.facebook.react.modules.core.DeviceEventManagerModule.RCTDeviceEventEmitter;
 import com.fidelreactlibrary.adapters.abstraction.DataProcessor;
 
-public final class ResultAvailableEventEmitter implements DataProcessor<WritableMap> {
+public final class BridgeLibraryEventEmitter implements DataProcessor<ReadableMap> {
 
     private final ReactApplicationContext reactContext;
+    private final BridgeLibraryEvent eventToEmit;
 
-    public ResultAvailableEventEmitter(ReactApplicationContext reactContext) {
+    public BridgeLibraryEventEmitter(ReactApplicationContext reactContext, BridgeLibraryEvent eventToEmit) {
         this.reactContext = reactContext;
+        this.eventToEmit = eventToEmit;
     }
 
     @Override
-    public void process(WritableMap data) {
+    public void process(ReadableMap data) {
```

Comment:
Why is it readable map here and writable map in `adaptedConsentDetailsProcessor.process`? will it work?